### DISCUSSION TITLE
returntype keyword

### DISCUSSION THREAD

**Discussion by NigelBess** [dotnet/csharplang#6074] [Language Ideas]
Here is a common pattern:

      ExampleType ExampleFunction()
       {
            var outvar = new ExampleType();
            //functional code here
            return outvar;
        }
        
I would like to replace that with:

        ExampleType ExampleFunction()
        {
            var outvar = new returntype();
            //functional code here
            return outvar;
        }

returntype would be a literal equivalent to writing out the return type of the current scope.

The purpose of this proposal is prevent the unavoidable DRY principle violation of repeating the return type, as well as improving readability by conveying intent. (why is outVar an ExampleType? because that is the type being returned by the function)

**Comment by Phantonia** [comment]
But surely the type would matter at more places in the method. For example, whenever a method would be called on `outvar`, changing the return type might break that method call. So, when would changing the return type ever leave the method body intact - except for replacing the old type with the new type?

**Reply by NigelBess** [reply]
Regardless of whether changing the return type breaks existing code, the returntype keyword still saves the time required to change the type in one or more additional locations. I've edited the post to better describe my intent.

**Comment by HaloFour** [comment]
The return type of the method is unrelated to the type of any given local, so I don't see this as violating DRY.  I'd prefer to not have the language add a bunch of obscure aliases.

**Comment by TahirAhmadov** [comment]
I see the merit in this, but this is adds very little actual value; even if it's deemed a good thing, it would likely be at the very bottom of priority list. Also, why not just go `var outvar = new return();`? It may be possible to avoid adding a new keyword.
